### PR TITLE
check EditTask. 

### PR DESCRIPTION
**Pull request by lz2512**
# Task Description
Implemented EditTask but was unable to test due to AddTask not working (see issue #47 )

## Type of change
New feature (non-breaking change which adds functionality)

## Solution
Implemented in similar fashion to UpdatePermission

## Documentation
image
image
image
image

# Checklist:
- [✓] My code follows the style guidelines of this project
- [✓] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [✓ ] My changes generate no new warnings
- [✓ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Andrew-Kwok** [comment]
use the PR template

**Comment by lz2512** [comment]
> use the PR template

Updated comment

File: src/main/java/data_access/todoist/ApiDataAccessObject.java [hunk 1]
```diff
@@ -139,4 +140,32 @@ public ArrayList<Task> getTasks(String projectID) throws Exception {
             throw new Exception("error connectıng to todoıst: " + e.getMessage());
         }
     }
+
+    @Override
+    public void editTask(Task task) throws Exception {
```
Comment (Andrew-Kwok) [inline src/main/java/data_access/todoist/ApiDataAccessObject.java:145]: edit task without using taskId doesn't make sense

File: src/main/java/domains/task/use_case/add_task/AddTaskOutputBoundary.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package domains.task.use_case.add_task;
 
 public interface AddTaskOutputBoundary {
-    void prepareSuccessView(AddTaskOutputData taskID);
+    void prepareSuccessView(AddTaskOutputData task);
```
Comment (Andrew-Kwok) [inline src/main/java/domains/task/use_case/add_task/AddTaskOutputBoundary.java:4]: change to addTaskOutputData

**Pull request merged**